### PR TITLE
Feature/improved event routing

### PR DESCRIPTION
**Pull request by 0xCA5A**

### CODE REVIEW

File: application/app/modules/BackendModule.java [hunk 1]
```diff
@@ -3,28 +3,49 @@
...
-import javax.inject.Inject;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 
 public class BackendModule extends AbstractModule {
-
     private final Config backendConfig;
 
-    @Inject
```
Comment (rake5k) [inline application/app/modules/BackendModule.java:18]: Did this even work?

File: backend/src/main/java/hal/JbertControlButtonName.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package hal;
+
+public class JbertControlButtonName {
```
Comment (rake5k) [inline backend/src/main/java/hal/JbertControlButtonName.java]: Shall the control and playlistselections button positions and gpio names be configurable in future?

File: backend/src/main/java/event/router/SimplePlaylistLoad.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        configureGpioEventHandling();
+    }
+
+    private void configureGpioEventHandling() {
+        gpioService.registerGpiListener(RED.getGpioName(), new PlaylistLoadAction(mpdService, "red"));
+        gpioService.registerGpiListener(BLACK.getGpioName(), new PlaylistLoadAction(mpdService, "black"));
+        gpioService.registerGpiListener(GREEN.getGpioName(), new PlaylistLoadAction(mpdService, "green"));
+        gpioService.registerGpiListener(BLUE.getGpioName(), new PlaylistLoadAction(mpdService, "blue"));
+        gpioService.registerGpiListener(YELLOW.getGpioName(), new PlaylistLoadAction(mpdService, "yellow"));
```
Comment (rake5k) [inline backend/src/main/java/event/router/SimplePlaylistLoad.java]: This looks like being easily replaceable by a iteration over the enum values:

```
EnumSet.allOf(JbertPlaylistSelectButton.class).forEach(...)
```

File: backend/src/main/java/event/router/InitPlaylistByRfidTagSelectTrackByGpi.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        gpioService.registerGpiListener(RED.getGpioName(),
+                new PlayTrackByIndexAction(mpdService, 0));
+        gpioService.registerGpiListener(BLACK.getGpioName(),
+                new PlayTrackByIndexAction(mpdService, 1));
+        gpioService.registerGpiListener(GREEN.getGpioName(),
+                new PlayTrackByIndexAction(mpdService, 2));
+        gpioService.registerGpiListener(BLUE.getGpioName(),
+                new PlayTrackByIndexAction(mpdService, 3));
+        gpioService.registerGpiListener(YELLOW.getGpioName(),
+                new PlayTrackByIndexAction(mpdService, 4));
```
Comment (rake5k) [inline backend/src/main/java/event/router/InitPlaylistByRfidTagSelectTrackByGpi.java]: Probably replaceable by a loop, too?

**Review by rake5k — COMMENTED** [review]
:+1:

**Review by rake5k — APPROVED** [review]
Looks good!

File: application/app/modules/BackendModule.java (hunk 1 above)
Comment (0xCA5A) [reply application/app/modules/BackendModule.java:18]: That worked fine. Too many @Injects do not seem to be a problem...

File: backend/src/main/java/hal/JbertControlButtonName.java (hunk 2 above)
Comment (0xCA5A) [reply backend/src/main/java/hal/JbertControlButtonName.java]: Not sure yet, might be interesting as soon as there is more than one hardware :)
I've changed the class to a enum anyway.

File: backend/src/main/java/event/router/SimplePlaylistLoad.java (hunk 3 above)
Comment (0xCA5A) [reply backend/src/main/java/event/router/SimplePlaylistLoad.java]: nice! done!

File: backend/src/main/java/event/router/InitPlaylistByRfidTagSelectTrackByGpi.java (hunk 4 above)
Comment (0xCA5A) [reply backend/src/main/java/event/router/InitPlaylistByRfidTagSelectTrackByGpi.java]: nice! done!

**Pull request merged**